src/gc/gc.cpp

```diff
@@ -9645,7 +9645,7 @@ void gc_heap::adjust_ephemeral_limits ()
                  (size_t)ephemeral_low, (size_t)ephemeral_high))
 
     // This updates the write barrier helpers with the new info.
-    StompWriteBarrierEphemeral(!!IsSuspendEEThread());
+    StompWriteBarrierEphemeral(g_fEEInit || !!IsSuspendEEThread());
```

Comment:
`g_fEEInit` is CoreCLR VM specific variable. It should not be used in the GC - it will cause problems with sharing the GC with ProjectN/CoreRT.